### ISSUE TITLE
Generated Request Body for Path is using 422 Response

### ISSUE THREAD

**Issue by xcskier56** [exoego/rspec-openapi#130]
Hi There, 

I have an odd issue. I have a request `PATCH /api/app_configs/:id` where I have a successful example and a failed example. No matter what order they are run in, the overall body for the request is always to body of the failed example which isn't a great experience for developers viewing the docs. 

The generated swagger has the expected 200 & 422 responses but the overall request body for the example is always being pulled from the 422 test. I have tried re-ordering the tests in the spec file, deleting everything and rebuilding, but no luck.

Here is my test:

```ruby
# frozen_string_literal: true

require 'rails_helper'

module Api
  RSpec.describe 'AppConfigs', type: :request do
    let(:account) { create(:account, projects: [project]) }
    let(:project) { create(:project) }
    let(:headers) { { 'Content-Type': 'application/json', 'X-Project-Id': project.id } }

    before { set_current_account(account) }

    describe 'PATCH /api/app_configs/:id', openapi: { summary: 'Update the Current App Config', tags: ['AppConfig'] } do
      let(:update_params) do
        {
          interface: 'interface AppConfig { foo: string, bar: number }',
          config: { foo: 'bar', bar: 1 }
        }
      end

      it 'Updates the current app config' do
        patch("/api/app_configs/#{project.app_config.id}", params: { app_config: update_params }.to_json, headers: headers)

        expect(response).to have_http_status :ok

        app_config_json = json_response['app_config']
        expect(app_config_json['interface']).to eq update_params[:interface]
        expect(app_config_json['config'].symbolize_keys).to eq update_params[:config]
      end

      context 'with invalid params' do
        let(:update_params) { { interface: nil, config: nil } }

        it 'Returns an error' do
          patch("/api/app_configs/#{project.app_config.id}", params: { app_config: update_params }.to_json, headers: headers)

          expect(response).to have_http_status :unprocessable_entity
          expect(json_response['errors']).to include('interface')
        end
      end
    end
  end
end
```

And here is the generated swagger:

```yaml
---
openapi: 3.0.3
info:
  title: Vessel Platform API
  version: 1.0.0
  description: Vessel Platform API
servers: []
components:
  schemas:
    AppConfig:
      type: object
      properties:
        id:
          type: string
        config:
          type: object
          properties: {}
        interface:
          type: string
        project_id:
          type: string
        created_at:
          type: string
        updated_at:
          type: string
paths:
  "/api/app_configs/{id}":
    patch:
      summary: Update the Current App Config
      tags:
      - AppConfig
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: string
        example: ea4497f0-ea99-474b-ab69-1dbb7b7b91e7
      requestBody:
        content:
          application/json:
            schema:
              type: object
              properties:
                app_config:
                  type: object
                  properties:
                    interface:
                      type: string
                      nullable: true
                    config:
                      type: object
                      properties:
                        foo:
                          type: string
                        bar:
                          type: integer
                      items:
                        type: integer
                      nullable: true
            example:
              app_config:
                interface:
                config:
      responses:
        '200':
          description: Updates the current app config
          content:
            application/json:
              schema:
                type: object
                properties:
                  app_config:
                    type: object
                    properties:
                      id:
                        type: string
                      config:
                        type: object
                        properties:
                          foo:
                            type: string
                          bar:
                            type: integer
                      interface:
                        type: string
                      project_id:
                        type: string
                      created_at:
                        type: string
                      updated_at:
                        type: string
              example:
                app_config:
                  id: 1c75a8e9-f32e-4913-bf34-585f5b78ad1c
                  config:
                    foo: bar
                    bar: 1
                  interface: 'interface AppConfig { foo: string, bar: number }'
                  project_id: 1100ef21-f88c-4dc8-89a7-4938a03a819a
                  created_at: '2023-08-16T15:03:13Z'
                  updated_at: '2023-08-16T15:03:13Z'
        '422':
          description: Returns an error
          content:
            application/json:
              schema:
                type: object
                properties:
                  errors:
                    type: object
                    properties:
                      interface:
                        type: array
                        items:
                          type: string
              example:
                errors:
                  interface:
                  - can't be blank
```

Here's a screenshot of the swagger UI

![Screenshot 2023-08-16 at 10 08 21 AM](https://github.com/exoego/rspec-openapi/assets/2357871/8e606d2b-47a6-49a8-bf6c-9426896f350b)

**Comment by exoego** [comment]
We may need an explicit way to specify which request body should be used as an overall example.

**Comment by xcskier56** [comment]
I think an API like that would be quite helpful. I have to structure my tests with `context` blocks in a specific way to get examples to show up properly. I'm happy to look into making a PR, could you give me some pointers to where to look to implement this?

**Comment by exoego** [comment]
I assume this issue is a duplicate of #140 and fixed by #154
Request bodies from failure examples (`4xx` and `5xx`) are no longer merged into `requestBody`.
Hope it fixes your problem.
